src/settings/components/SettingBlocks.tsx

```diff
@@ -310,10 +315,33 @@ const ModelSettingsComponent: React.FC<ModelSettingsComponentProps> = ({
     }
   };
 
+  const handleVerify = async () => {
+    if (newModel.name && newModel.provider) {
+      setIsVerifying(true);
+      await ping({ ...newModel })
+        .then((_) => new Notice("Model verification successful!"))
+        .catch((err) => {
+          const errorMessage = "Model verification failed: " + err2String(err);
+          console.error(err);
+          new Notice(errorMessage);
+        });
+
+      setIsVerifying(false);
+    } else {
+      new Notice("Please fill in necessary fields!");
+    }
+  };
+
   const handleSetDefaultModel = (model: CustomModel) => {
     onSetDefaultModelKey(getModelKey(model));
   };
 
+  const showBaseUrlOption = !isEmbeddingModel
```

Comment:
This should not be here. We don't want to couple everything. Model-related stuff should be in model-related files.